### PR TITLE
fix icon menu content placement

### PR DESCRIPTION
**Pull request by warm-coolguy**
## Summary

- Replaced hack with absolute positioning.
- Fixed bug that appeared when Zoom was sorted prior to Attributions, but decided to not render itself due to tiny client window.
- Remove spacing on IconMenu's ul that seems to have leaked in unnoticed.

## Instructions for local reproduction and review

Run arbitrary clients in arbitrary environments to check if something is broken. Most notable change should occur in Meldemichel example clients for `'SINGLE'` mode, where the attributions were previously displayed above the map.

### CODE REVIEW

File: packages/plugins/IconMenu/src/components/IconMenu.vue [hunk 1]
```diff
@@ -112,31 +98,33 @@ export default Vue.extend({
...
+  top: 0;
+  right: calc(100% + 0.5em);
 }
 
 .icon-menu-list-item-content-horizontal {
   position: absolute;
   white-space: nowrap;
-  top: 3em;
+  top: calc(100% + 0.5em);
+  right: -0.5em;
```
Comment (dopenguin) [inline packages/plugins/IconMenu/src/components/IconMenu.vue:127]:
```suggestion
  right: 0;
```
Otherwise it hugs the side of the display

**Review by dopenguin — CHANGES_REQUESTED** [review]
Content like the one from the attribution plugin is still overflowing outside of the client.  
Should this be handled as a separate task or would you like to tackle this here?

🏓@warm-coolguy

File: packages/plugins/IconMenu/src/components/IconMenu.vue (hunk 1 above)
Comment (warm-coolguy) [reply packages/plugins/IconMenu/src/components/IconMenu.vue:127]: I figured it was more important to have the box align to the buttons than to have the sidebar not hugging the side of the viewport.

spacing
spacing-with-sidebar

**Comment by warm-coolguy** [comment]
> Content like the one from the attribution plugin is still overflowing outside of client. Should this be handled as a separate task or would you like to tackle this here?
> 
> 🏓@warm-coolguy

I think it's acceptable for IconMenu contents to overflow the client. After all, such windows can be closed (and hence don't prevent using other overlapped contents), and especially on narrow use cases (i.e. the sidebar scenario) there's hardly enough space inside the client for comfortable/usable menus.

However, I agree that the contents should always be reachable. On writing an issue, I failed to create a scenario where the contents left the reachable area. Are you able to create such a scenario? If so, please do make an issue of it.

🏓 @dopenguin

File: packages/plugins/IconMenu/src/components/IconMenu.vue (hunk 1 above)
Comment (dopenguin) [reply packages/plugins/IconMenu/src/components/IconMenu.vue:127]: When testing with `right: 0;` on a phone with a viewport of `412px x 919px`, I received the following results in Chrome:

att_0
layerchooser_0

With `right: -0.5em;` the results were the following:

att_-0 5em
layerchooser_-0 5em

Meaning, that the alignment with the Buttons was only present when using `right: 0;`.

Could you please elaborate in which use cases the above situations would occur?

**Comment by dopenguin** [comment]
> > Content like the one from the attribution plugin is still overflowing outside of client. Should this be handled as a separate task or would you like to tackle this here?
> > 🏓@warm-coolguy
> 
> I think it's acceptable for IconMenu contents to overflow the client. After all, such windows can be closed (and hence don't prevent using other overlapped contents), and especially on narrow use cases (i.e. the sidebar scenario) there's hardly enough space inside the client for comfortable/usable menus.
> 
> However, I agree that the contents should always be reachable. On writing an issue, I failed to create a scenario where the contents left the reachable area. Are you able to create such a scenario? If so, please do make an issue of it.
> 
> 🏓 @dopenguin

Tackled the issue in #17

File: packages/plugins/IconMenu/src/components/IconMenu.vue (hunk 1 above)
Comment (warm-coolguy) [reply packages/plugins/IconMenu/src/components/IconMenu.vue:127]: `right: 0;` visibly doesn't produce alignment between the buttons and the contents in your screenshots.

My screenshots have been provided from desktop browsers. I see, the difference is that `scrollbar-gutter` produces the right distance in desktop browsers, but not on mobile devices, where scrollbars are usually these thin lines atop the inside of boxes.

For the time being, I suggest we leave it at that. It looks good on desktop and acceptable on mobile devices. Let's say it's intentional to make better usage of the available space. With `right: 0;`, both are slightly off.

Comment (dopenguin) [reply packages/plugins/IconMenu/src/components/IconMenu.vue:127]: We'll probably get back to this, but fine by me for the time being

**Review by dopenguin — APPROVED** [review]
🚀 @warm-coolguy

**Pull request merged**